### PR TITLE
[GraphBolt] Use indexing in DistributedItemSampler

### PR DESCRIPTION
**Pull request by RamonZhou**
## Description
Using indexing method in DIS.
The doc string hasn't finished yet.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/item_sampler.py [hunk 1]
```diff
@@ -167,19 +176,102 @@ def _calculate_offsets(self, buffer):
...
+            batch_remain = big_batch_remain % self._batch_size
+            last_batch_count = big_batch_remain // self._batch_size
+            if self._rank < last_batch_count:
+                last_batch = self._batch_size
+            elif self._rank == last_batch_count:
+                last_batch = batch_remain
+            else:
+                last_batch = 0
+            num_items = big_batch_count * self._batch_size + last_batch
+            start_offset = (
```
Comment (peizhou001) [inline python/dgl/graphbolt/item_sampler.py:204]: Is this correct? why it start in the last big batch

File: python/dgl/graphbolt/item_sampler.py [hunk 2]
```diff
@@ -167,19 +176,102 @@ def _calculate_offsets(self, buffer):
...
+                big_batch_count * self._batch_size * self._rank
+                + min(self._rank * self._batch_size, big_batch_remain)
+            )
+            if not self._drop_uneven_inputs or (
```
Comment (peizhou001) [inline python/dgl/graphbolt/item_sampler.py:208]: Add description here

File: python/dgl/graphbolt/item_sampler.py [hunk 3]
```diff
@@ -167,19 +176,102 @@ def _calculate_offsets(self, buffer):
...
+                not self._drop_last and last_batch_count == self._num_replicas
+            ):
+                num_evened_items = num_items
+                if num_workers > 1:
```
Comment (peizhou001) [inline python/dgl/graphbolt/item_sampler.py:212]: Roughly looking there are a lot of repeat code here, try to merge them and make the code cleaner.

**Review by frozenbugs — APPROVED** [review]

**Comment by RamonZhou** [comment]
@dgl_bot

**Pull request merged**